### PR TITLE
Mantas tdd-cohort-manager-challenge

### PR DESCRIPTION
**Pull request by ManNavic**

### CODE REVIEW

File: package-lock.json
Comment (marc-alzapiedi) [inline package-lock.json:1]: All the changes to the package.json file, I would like to know what command line you ran to install these dependencies, and why? Is it becuase you are using the Twilio?

Have you run?
``` 
npm ci
```

Comment (ManNavic) [reply package-lock.json:1]: npm install twilio , yes it's for Twilio

File: spec/cohort.spec.js [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    })
+  })
+  describe('addStudentToCohort', () => {
+    it('Add a student to a specific cohort', () => {
+      const expected = [
+        {
+          id: 1,
+          firstName: 'John',
+          lastName: 'Cena',
+          gitHub: 'JohnCena',
```
Comment (marc-alzapiedi) [inline spec/cohort.spec.js:75]: HHAHAHAAHA JOHN CENA

File: src/cohort.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    if (this.cohortList.length === 0) {
+      throw new Error('ERROR: Cohort List is empty')
+    }
+
+    const targetCohort = this.cohortList.find((cohort) => {
+      const cohortNameList = Object.keys(cohort)[0]
+      return cohortNameList === cohortName
+    })
+
+    if (!targetCohort) {
```
Comment (marc-alzapiedi) [inline src/cohort.js:33]: Is the value of targetCohort true or false? Also like the use of find, wouldn't think of using it like that

File: src/cohort.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
+const Twilio = require('../src/sms.js')
+class Cohorts {
+  constructor() {
+    this.cohortList = []
```
Comment (marc-alzapiedi) [inline src/cohort.js:4]: It looks like you are only using one class and putting everything in this array as objects?

File: spec/cohort.spec.js [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+    })
+    it('Cohort list is empty', () => {
+      const expected = []
+      const result = cohortLi.cohortList
+      expect(result).toEqual(expected)
+    })
+  })
+  describe('searchByName', () => {
+    it('Search cohort by name', () => {
+      const expected = { 'Cohort 1': [] }
```
Comment (marc-alzapiedi) [inline spec/cohort.spec.js:33]: It this your structure of your local datastore?

```
this.cohortList = []
```

I also have a similar structure where I have an array as a value to an object key, and fill it with students belonging to a cohort.

File: src/sms.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      .create({
+        body: `\nCongratulations ${studentName}\nYou have been added to ${cohortName}\n You are Student in ${cohortName} from :${regTime}`,
+        from: '+15154978934',
+        to: `${contactNumber}`
+      })
+      .then((message) => console.log(message.status))
+  }
+}
+
+module.exports = Twilio
```
Comment (marc-alzapiedi) [inline src/sms.js:19]: Need to try experiementing with twilio

File: src/cohort.js [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    this.id = 1
+  }
+
+  addCohort(cohortName) {
+    if (cohortName === '') {
+      throw new Error('Error: Please provide a Cohort name')
+    }
+    for (const cohort of this.cohortList) {
+      const cohortNameList = Object.keys(cohort)[0]
```
Comment (sarajbeazley) [inline src/cohort.js:13]: I'm not familiar with this Object.keys method, need to look into this

File: src/cohort.js [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+        this.cohortList.splice(index, 1)
+        console.log(this.cohortList)
+        return this.cohortList
+      } else if (cohortNameList !== cohortName) {
+        throw new Error('Error: Cohort name does not exist')
+      }
+    }
+  }
+
+  addStudentToCohort(
```
Comment (sarajbeazley) [inline src/cohort.js:58]: I used just the name and ID to add new student - perhaps I should have also included all the rest of the info too

**Pull request closed without merging**